src/components/album/AlbumChat.js

```diff
@@ -53,6 +58,13 @@ const AlbumChat = ({ apiUrl }) => {
     <div>
       <h2>Album Chat</h2>
       <div>
+        <div>
+          {messages.map((message, index) => (
+            <div key={index}>
```

Comment:
⚠️  This is a common pitfall in React 😢 

The official React docs explain it better than me: https://react.dev/learn/rendering-lists

But basically, the issue is that every time we add a new message to the state (ie: with `setMessages`), React re-renders this component, that means the `index` that was there before is now going to be a new `index` (even if it ends up being the same number, it'll be a new "instance" of that number because React is going to do `messages.map` again). 

Instead, React wants us to use a value that it can keep track of across renders. 

In this case, we can handle this in the frontend or the backend (or a mix of the two):
**Handle in the frontend**
For the messages we get from the `'chat:album'` socket listener, we can add an "key" property to the `data` payload, with the official doc's suggestion of `crypto.randomUUID()` to generate the key. This can then be used as the key here when we iterate over them

**Handle in the backend**
If the socket handler in the backend is going to save `ChatMessage` to the database anyway when it receives them, then it can return that mongo object back when it emits a message; that will have a mongo `_id` field that we know is unique and we safely use that field as the iteration key here in the React code.

Reply:
Since in this case, there's already work being done by you on the backend to save the ChatMessage when receiving these messages, then that is likely the best road to go down. But wanted to present both options just for learning purposes that this can be done even if the object you're iterating over is not something that is being saved in the db.

Reply:
I made a little temporary design just to see how it would turn out. But later, we will redo the entire design when we decide exactly how it will look full app.

Reply:
<img width="643" alt="11" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/95896159/e18a9a6f-0718-472e-8fdb-dc12ff0c82f2">

Reply:
On the backend, I'm receiving a response like the one in the screenshot. However, on the frontend, I'm trying to display the username:message properly instead of :message. But not work for me

Reply:
This initial design looks fantastic Marina, great job with it!